app/api/suggestions/suggestions.ts

```diff
@@ -339,4 +346,179 @@ export const Suggestions = {
+      const propNames: string[] = deletedTemplatesAndDeletedTemplateProps.reduce(
+        (acc, curr) => [...acc, ...curr.properties],
+        []
+      );
+      const uniquePropNames: string[] = [...new Set<string>(propNames)];
+
+      await IXSuggestionsModel.db.deleteMany({
+        $and: [
+          { entityId: { $in: entitiesSharedIds } },
+          { propertyName: { $in: uniquePropNames } },
+        ],
+      });
+    }
+  },
+
+  getTemplatesWithNewProps: (
+    settingsTemplates: ISettingsTemplate[],
+    currentSettingsTemplates: ISettingsTemplate[]
+  ) =>
+    settingsTemplates
+      .map(newTemp => {
+        const oldTemplate = currentSettingsTemplates?.find(
+          oldTemp => oldTemp.template === newTemp.template
+        );
+        if (newTemp.properties.length === oldTemplate?.properties.length || !oldTemplate) {
+          return null;
+        }
+        const newProps = newTemp.properties;
+        const oldProps = oldTemplate?.properties || [];
+        const addedProps: string[] = newProps
+          .map((prop: any) => (!oldProps.includes(prop) ? prop : false))
+          .filter(p => p);
+        return { ...newTemp, properties: addedProps };
+      })
+      .filter(t => t),
+
+  getTemplateDifference: (
+    currentSettingsTemplates: ISettingsTemplate[],
+    settingsTemplates: ISettingsTemplate[]
+  ) => {
+    const newTemplates = settingsTemplates.filter(temp => {
+      const oldTemplateIds = currentSettingsTemplates?.map(oldTemp => oldTemp.template) || [];
+      return !oldTemplateIds.includes(temp.template);
+    });
+
+    const combedNewTemplates = Suggestions.getTemplatesWithNewProps(
+      settingsTemplates,
+      currentSettingsTemplates
+    );
+
+    return newTemplates.concat(combedNewTemplates as ISettingsTemplate[]);
+  },
+
+  saveConfigurations: async (settingsTemplates: ISettingsTemplate[]) => {
+    const currentSettings = await settings.get();
+    const defaultLanguage = currentSettings?.languages?.find(lang => lang.default)?.key;
+    let currentSettingsTemplates: ISettingsTemplate[] | undefined =
+      currentSettings.features?.metadataExtraction?.templates;
+    currentSettingsTemplates = currentSettingsTemplates || [];
+
+    await Suggestions.deleteSuggestionsNotConfigured(currentSettingsTemplates, settingsTemplates);
+    // @ts-ignore
+    currentSettings.features.metadataExtraction.templates = settingsTemplates;
+    await settings.save(currentSettings);
+
+    const newTemplates = Suggestions.getTemplateDifference(
+      currentSettingsTemplates,
+      settingsTemplates
+    );
+    await Suggestions.createBlankState(newTemplates, defaultLanguage as string);
+
+    return currentSettings;
+  },
+
+  fetchEntitiesBatch: async (query: any, limit: number = 100) =>
+    entitiesModel.db.find(query).select('sharedId').limit(limit).sort({ _id: 1 }).exec(),
+
+  createBlankState: async (settingsTemplates: any[], defaultLanguage: string) => {
```

Comment:
It looks like this shouldn't be exported either.

Reply:
This one is most likely exported since it is tested separately, so you would need to export it anyway. I think it is fine to leave it inside the object.